incapsula/resource_api_client.go

```diff
@@ -0,0 +1,235 @@
+package incapsula
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceApiClient() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceApiClientCreate,
+		Read:   resourceApiClientRead,
+		Update: resourceApiClientUpdate,
+		Delete: resourceApiClientDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the API client.",
+			},
+			"expiration_period": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Expiration period for the API key (RFC3339 or duration, e.g. 30d).",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Whether the API client is enabled.",
+			},
+			"grace_period": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Grace period in seconds.",
+			},
+			"regenerate_version": {
+				Type:        schema.TypeFloat,
+				Optional:    true,
+				Description: "Increment to trigger API key regeneration.",
+			},
+			"api_key": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+				Description: "Generated API key for client authentication.",
+			},
+			"api_client_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the API client.",
+			},
+			"expiration_date": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Expiration date of the API key.",
+			},
+			"last_used_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Last used timestamp.",
+			},
+		},
+	}
+}
+
+// PATCH /v3/api-client/{client_id} for create, update, enable/disable, regenerate
+func resourceApiClientCreate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
We still don't have the API of 'create', so this method must return an error: 'creation of an API Client is not supported yet'. Implement this method as part of the API_client creation epic later in the quarter.